Make arcade.Text lazy

csd4ni3l opened the pull request:
This is a draft to make arcade.Text lazy.

It works in the same way as the lazy NinePatch, using the `_initialized` attribute and the `initalize` and `_init_deferred` functions. It raises a `RuntimeError` if a property of the label is accessed before initializing.

I dont know how to handle arguments correctly or efficiently, so its marked as a draft, but in its current state, its still usable.

csd4ni3l committed:
Make arcade.Text lazy

csd4ni3l committed:
Fix size function return type

csd4ni3l committed:
Try to fix size function return type

csd4ni3l committed:
Make code lines shorter to fix typing errors

csd4ni3l committed:
remove trailing whitespace and fix type of size function

einarf commented on arcade/text.py:
```diff
@@ -276,8 +276,10 @@ def __init__(
+        self._initialized = False
+        self.arguments = [text, x, y, color, font_size, width, align, font_name, bold,
+                          italic, anchor_x, anchor_y, multiline, rotation, batch, group, z]
+        self.kwargs = kwargs
```
Should probably be a private member

einarf commented on arcade/text.py:
```diff
@@ -288,54 +290,82 @@ def __init__(
         """
-        self._label.begin_update()
+        if self._initialized:
+            self._label.begin_update()
```
Instead of checking for initialized everywhere could we make a `label` property that returns the underlying pyglet label instead and only have the initialized check there?

csd4ni3l replied:
Good idea. Let me do that as soon as i can.

csd4ni3l committed:
Add the label property and make arguments private

- Add the label property and use it in the code, if its not initialized, raise a RuntimeError.
- Make the `arguments` and `kwargs` private, using _

csd4ni3l committed:
Fix label property errors

csd4ni3l committed:
Fix docstring and formatting

csd4ni3l committed:
Fix formatting

csd4ni3l committed:
Only initialize label if its not already

csd4ni3l committed:
Use a dict for both args and kwargs

csd4ni3l committed:
Fix font_name typing issue

csd4ni3l committed:
fix pyglet.text.Label typing issue
